corehq/apps/case_importer/tracking/filestorage.py

```diff
@@ -0,0 +1,94 @@
+import os
+from tempfile import mkstemp
+import uuid
+from django.core.cache import caches
+from corehq.apps.case_importer.tracking.models import CaseUploadFileMeta
+from corehq.blobs import get_blob_db
+from corehq.blobs.util import random_url_id
+from corehq.util.files import file_extention_from_filename
+from dimagi.utils.decorators.memoized import memoized
+
+BUCKET_PREFIX = 'case_importer'
+
+
+class PersistentFileStore(object):
+    def __init__(self, bucket, meta_model):
+        """
+        :meta_model is a django model used to store meta info
+            must contain columns identifier, filename, length
+        """
+
+        self._bucket_prefix = bucket
+        self._db = get_blob_db()
+        self._meta_model = meta_model
+
+    def _get_bucket(self, padding):
+        return '{}/{}'.format(self._bucket_prefix, padding)
+
+    def write_file(self, f, filename):
+        padding = random_url_id(16)
+        blob_info = self._db.put(f, bucket=self._get_bucket(padding))
+        identifier = '{}/{}'.format(padding, blob_info.identifier)
+        self._meta_model(identifier=identifier, filename=filename,
+                         length=blob_info.length).save()
+        return identifier
+
+    @memoized
+    def get_tempfile(self, identifier):
```

Comment:
Yeah I like that suggestion. This was modeled after the maybe even-more-poorly-named `DownloadBase.get_filename()`, which does essentially the same thing. I think maybe most correct would be `get_tempfile_ref_for_contents`; I'll go with that even though it might be a mouth full.